packages/sourcecred/src/cli/grain.js

```diff
@@ -45,6 +46,25 @@ const grainCommand: Command = async (args, std) => {
 
   const {distributions, ledger} = await grain(grainInput);
 
+  const integrationCurrency = grainInput.currencyDetails.integrationCurrency;
+  const grainIntegration = grainInput.grainConfig.integration;
+  if (integrationCurrency && grainIntegration) {
+    distributions.forEach((distribution) => {
+      const result = executeGrainIntegration(
+        ledger,
+        grainIntegration.function,
+        distribution,
+        integrationCurrency,
+        false
+      );
+      if (result.grainIntegrationOutput)
+        instance.writeGrainIntegrationOutput(
+          result.grainIntegrationOutput,
+          distribution
+        );
+    });
+  }
```

Comment:
Shouldn't this go in the grain API instead of the grain CLI since it is business logic?

Reply:
good point. I'll separate the business logic from the I/O